src/integration-tests/debugClient.ts

```diff
@@ -371,6 +371,24 @@ export class CdtDebugClient extends DebugClient {
         });
         return evalResponse.body.result;
     }
+
+    /**
+     * Obtain the output coming from the debug console..
+     * @param category catgory of output event coming on debug console.
+     * @param output expected output coming on debug console
+     */
+    public async getDebugConsoleOutput(
+        launchArgs: any,
+        category: string,
+        output: string
+    ): Promise<any> {
+        return Promise.all([
+            this.waitForEvent('initialized'),
+            this.launch(launchArgs).then((_event) => {
+                return this.waitForOutputEvent(category, output);
+            }),
+        ]);
```

Comment:
> Is there anything I'm missing?

The reason the tests are unstable is because you only start waiting for the output after launch returns - but if the output comes too quickly after the launch before you have a chance to start `waitForOutputEvent` the output you are looking for is missed. Therefore you have a race condition here.

Here is a suggested change. 

```suggestion
        return Promise.all([
            this.waitForEvent('initialized'),
            this.launch(launchArgs),
            this.waitForOutputEvent(category, output),
        ]);
```